test/suite/test_compact07.py

```diff
@@ -0,0 +1,212 @@
+
+        time.sleep(1)
+        
+        # Periodically check that foreground compaction has done some work on the small table.
+        pages_rewritten = self.get_pages_rewritten(uri_small)
+        while (pages_rewritten == 0):
+            time.sleep(1)
+            pages_rewritten = self.get_pages_rewritten(uri_small)
+                    
+        # Disable the background compaction server.
+        self.session.compact(None,'background=false')
+        
+        # Check that the background server is no longer running.
+        stat_cursor = self.session.open_cursor('statistics:', None, None)
+        compact_running = stat_cursor[stat.conn.background_compact_running][2]
+        self.assertEqual(compact_running, 0)
+        stat_cursor.close()
+    
+    # Test the background server while creating and dropping tables.
+    def test_background_compact_table_ops(self):
+        # Trigger the background compaction server first, before creating tables.
+        self.session.compact(None,'background=true,free_space_target=1MB')
+
+        # Create n tables for background compaction to loop through.
+        for i in range(self.n_tables):
+            uri = self.uri_prefix + f'_{i}'
+            ds = SimpleDataSet(self, uri, self.table_numkv, 
+                            key_format=self.key_format, 
+                            value_format=self.value_format)
+            ds.populate()
+                
+        # Delete some data from the half the tables.
+        for i in range(self.n_tables):
+            if (i % 2 == 0):
+                self.delete_range(f'{self.uri_prefix}_{i}')
+                
+        # Periodically check that background compaction has done some work on the first table.
+        pages_rewritten = self.get_pages_rewritten(f'{self.uri_prefix}_0')
+        while (pages_rewritten == 0):
+            time.sleep(1)
+            pages_rewritten = self.get_pages_rewritten(f'{self.uri_prefix}_0')
+            
+        # Now drop all the tables.
+        for i in range(self.n_tables):
+            self.dropUntilSuccess(self.session, f'{self.uri_prefix}_{i}')
+        
+        time.sleep(2)        
+        
+    # Run background compaction alongside many checkpoints.
+    def test_background_compact_chkpt_stress(self):
+        # Create n tables for background compaction to loop through.
+        for i in range(self.n_tables):
+            uri = self.uri_prefix + f'_{i}'
+            ds = SimpleDataSet(self, uri, self.table_numkv, 
+                            key_format=self.key_format, 
+                            value_format=self.value_format)
+            ds.populate()
+            self.delete_range(uri)
+            
+        self.reopen_conn()
+
+        self.session.compact(None, 'background=true,free_space_target=1MB')    
+        for i in range(1000):
+            self.session.begin_transaction()
+            cur = self.session.open_cursor(self.uri_prefix + '_0')
```

Comment:
The prefix is constant. We could define it outside of the loop or change the URI as we loop through the iterations. Since it is a stressing test, I wonder if we should have a cpp test instead.

Reply:
I think that will be worthwhile, let's make another ticket for that.

Reply:
We have WT-11346 for it, I will refer to those lines in the ticket.
If you decide to leave the test, please add a fixme.

Reply:
Thank you!